source/speech.py

```diff
@@ -1161,7 +1161,17 @@ def getFormatFieldSpeech(attrs,attrsCache=None,formatConfig=None,unit=None,extra
 		oldColor=attrsCache.get("color") if attrsCache is not None else None
 		backgroundColor=attrs.get("background-color")
 		oldBackgroundColor=attrsCache.get("background-color") if attrsCache is not None else None
-		if color and backgroundColor and color!=oldColor and backgroundColor!=oldBackgroundColor:
+		backgroundColorTwo=attrs.get("background-colorTwo")
+		oldBackgroundColorTwo=attrsCache.get("background-colorTwo") if attrsCache is not None else None
+		backgroundPattern=attrs.get("background-pattern")
+		oldBackgroundPattern=attrsCache.get("background-pattern") if attrsCache is not None else None
+		if color and backgroundColor and backgroundColorTwo and color!=oldColor and backgroundColor!=oldBackgroundColor and backgroundColorTwo!=oldBackgroundColorTwo:

```

Comment:
I think there are some cases that aren't covered here. This condition requires all of the three colours (foreground, background and background 2) to change in order for this to be reported. What if only one of the background colours changes? Or what if the foreground colour changes but the other two don't?

I don't think we want to have an elif for every single possibility. Rather, I'd combine the two background colours into one translated string earlier (if there are two colours). Then, you just check for a change in either backgroundColor or backgroundColor2.